src/rules/no-extraneous-dependencies.js

```diff
@@ -102,9 +111,9 @@ module.exports.schema = [
   {
     'type': 'object',
     'properties': {
-      'devDependencies': { 'type': 'boolean' },
-      'optionalDependencies': { 'type': 'boolean' },
-      'peerDependencies': { 'type': 'boolean' },
+      'devDependencies': { 'type': ['boolean', 'string', 'object'] },
+      'optionalDependencies': { 'type': ['boolean', 'string', 'object'] },
+      'peerDependencies': { 'type': ['boolean', 'string', 'object'] },
```

Comment:
You're allowing `object`, because a regular expression value is an object, right? Just want confirmation.